lib/graphql/query.rb

```diff
@@ -6,17 +6,43 @@ def initialize(names)
     end
   end
 
+  class InputValidationResult
```

Comment:
how about putting this in `lib/graphql/query/input_validation_result.rb`, just to keep `query.rb` a bit smaller?